extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchAggregatorTest.java

```diff
@@ -417,6 +440,153 @@ public void testPostAggs() throws Exception
+                                                             .withDimensionsSpec(NestedDataTestUtils.AUTO_DISCOVERY)
+                                                             .withMetrics(aggs)
+                                                             .withQueryGranularity(Granularities.NONE)
+                                                             .withRollup(true)
+                                                             .withMinTimestamp(0)
+                                                             .build()
+                                   )
+                                   .inputSource(
+                                       ResourceInputSource.of(
+                                           NestedDataTestUtils.class.getClassLoader(),
+                                           NestedDataTestUtils.ARRAY_TYPES_DATA_FILE
+                                       )
+                                   )
+                                   .inputFormat(NestedDataTestUtils.DEFAULT_JSON_INPUT_FORMAT)
+                                   .transform(TransformSpec.NONE)
+                                   .inputTmpDir(groupByFolder.newFolder());
+
+    List<Segment> realtimeSegs = ImmutableList.of(
+        new IncrementalIndexSegment(bob.buildIncrementalIndex(), SegmentId.dummy("test_datasource"))
+    );
+    List<Segment> segs = ImmutableList.of(
+        new QueryableIndexSegment(bob.buildMMappedMergedIndex(), SegmentId.dummy("test_datasource"))
+    );
+
+    GroupByQuery query = GroupByQuery.builder()
+                                     .setDataSource("test_datasource")
+                                     .setGranularity(Granularities.ALL)
+                                     .setInterval(Intervals.ETERNITY)
+                                     .setAggregatorSpecs(
+                                         new HllSketchBuildAggregatorFactory("a0", "arrayString", null, null, null, false, false, false),
+                                         new HllSketchBuildAggregatorFactory("a1", "arrayLong", null, null, null, false, false, false),
+                                         new HllSketchBuildAggregatorFactory("a2", "arrayDouble", null, null, null, false, false, false),
+                                         new HllSketchMergeAggregatorFactory("a3", "hll0", null, null, null, false, false),
+                                         new HllSketchMergeAggregatorFactory("a4", "hll1", null, null, null, false, false),
+                                         new HllSketchMergeAggregatorFactory("a5", "hll2", null, null, null, false, false),
+                                         new HllSketchMergeAggregatorFactory("a6", "hll3", null, null, null, false, false),
+                                         new HllSketchMergeAggregatorFactory("a7", "hll4", null, null, null, false, false),
+                                         new HllSketchMergeAggregatorFactory("a8", "hll5", null, null, null, false, false),
```

Comment:
`processAsArrays` has no impact at query time because we have column type information, the `processAsArrays` tests are the ingest time aggs creating the pre-agg sketches